src/p_map.c

```diff
@@ -1554,13 +1555,35 @@ static boolean PTR_ShootTraverse(intercept_t *in)
 	  // it's a sky hack wall
 	  // fix bullet-eaters -- killough:
 	  if  (li->backsector && li->backsector->ceilingpic == skyflatnum)
-	    if (demo_compatibility || li->backsector->ceilingheight < z)
-	      return false;
+    	  {
+      	    if (li->backsector->ceilingheight < z)
+              return false;
+            else if (demo_compatibility)
+              safe = true;
+    	  }
 	}
+      if(CRITICAL(aimslope))
```

Comment:
We don't have direct vertical aiming, so there is close to zero chance to accidentally shoot through a floor or ceiling. We strictly don't need this part of the code.